app/models/course/experience_points_record.rb

```diff
@@ -30,4 +32,34 @@ def manually_awarded?
   def reason
     manually_awarded? ? super : specific.experience_points_display_reason
   end
+
+  private
+
+  def send_notification
+    Course::LevelNotifier.level_reached(course_user.user, level_after_update)
+  end
+
+  # Test if the course_user will reach a new level after current update.
+  def reached_new_level?
+    return false if points_awarded.nil? || !points_awarded_changed?
```

Comment:
In a weird way, it seems more readable if we use unless here -> 
`return false unless points_awarded && points_awarded_changed?`